App/Sagas/TextileSagas.ts

```diff
@@ -93,26 +94,43 @@ export function * updateNodeOverview ( action: ActionType<typeof TextileNodeActi
   }
 }
 
+export function * onboardedSuccess(action: ActionType<typeof PreferencesActions.onboardedSuccess>) {
+  let defaultThread: TT.Thread = yield call(getDefaultThread)
+  if (!defaultThread) {
+    // Skipping the normal addThread request here because we don't
+    // want to redirect to the default thread after complete
+    const thread: Thread = yield call(TextileNode.addThread, 'default' as TT.ThreadName)
+    yield put(ThreadsActions.addThreadSuccess(thread))
+    yield put(ThreadsActions.refreshThreadsRequest())
+  }
+}
+
 export function * handleProfilePhotoSelected(action: ActionType<typeof UIActions.selectProfilePicture>) {
   yield put(PreferencesActions.onboardedSuccess())
   yield call(NavigationService.navigate, 'PrimaryNavigation')
-  yield take(getType(TextileNodeActions.startNodeSuccess))
 
-  let defaultThread: TT.Thread | undefined = yield call(getDefaultThread)
-  if (!defaultThread) {
-    yield put(ThreadsActions.addThreadRequest('default' as TT.ThreadName))
-    const action: ActionType<typeof ThreadsActions.addThreadSuccess> = yield take(getType(ThreadsActions.addThreadSuccess))
-    defaultThread = action.payload.thread
-    yield put(ThreadsActions.refreshThreadsRequest())
+  // PreferencesActions.onboardedSuccess will setup the node/default thread, so wait for those
```

Comment:
Seems like something would probably be very broken if the node weren't online at this point. If that were the case, your loop that waits for it to come online will probably be looping forever. We should probably disable the UI for updating your avatar if the node isn't online.